lib/Frontend/ParseableInterfaceModuleLoader.cpp

```diff
@@ -559,7 +570,14 @@ class swift::ParseableInterfaceBuilder {
       if (ShouldSerializeDeps)
         SerializationOpts.Dependencies = Deps;
       SILMod->setSerializeSILAction([&]() {
-        serialize(Mod, SerializationOpts, SILMod.get());
+        if (ModuleBuffer) {
+          // We don't want to serialize module docs in the cache -- they
+          // will be serialized beside the interface file.
+          serializeToBuffers(Mod, SerializationOpts, ModuleBuffer,
+                             /*ModuleDocBuffer*/nullptr, SILMod.get());
```

Comment:
Seems like we could always pass down `ModuleBuffer` and `nullptr` and get the same effect?

Reply:
I thought about that, but I preferred explicitly encoding the difference in behavior. But if you think they should be combined, I'll combine them.

Reply:
I feel like the less a function changes its behavior based on inputs, the better.